src/components/util/dnd-table.js

```diff
@@ -0,0 +1,270 @@
+                    : handleClickIfUnchecked();
+            }}
+        />
+    );
+}
+
+function DefaultTableCell({ arrayFormName, rowIndex, column }) {
+    const valueToWatch = useWatch({
+        name: `${arrayFormName}[${rowIndex}].${column.dataKey}`,
+    });
+    return <TableCell key={column.dataKey}>{valueToWatch}</TableCell>;
+}
+
+function EditableTableCell({ arrayFormName, rowIndex, column }) {
+    return (
+        <TableCell key={column.dataKey}>
+            <TableNumericalInput
+                name={`${arrayFormName}[${rowIndex}].${column.dataKey}`}
+            />
+        </TableCell>
+    );
+}
+
+const DndTable = ({
+    arrayFormName,
+    useFieldArrayOutput,
+    disabled,
+    columnsDefinition,
+    handleAddButton,
+    handleDeleteButton,
+    handleUploadButton,
+    uploadButtonMessageId,
+}) => {
+    const intl = useIntl();
+
+    const { getValues, setValue } = useFormContext();
+
+    const {
+        fields: tapFields, // don't use it to access form data ! check doc
+        move,
+        swap,
+    } = useFieldArrayOutput;
+
+    function renderTableCell(rowId, rowIndex, column) {
+        let CustomTableCell = column.editable
+            ? EditableTableCell
+            : DefaultTableCell;
+        return (
+            <CustomTableCell
+                key={rowId + column.dataKey}
+                arrayFormName={arrayFormName}
+                rowIndex={rowIndex}
+                column={column}
+            />
+        );
+    }
+
+    function selectAllRows() {
+        for (let i = 0; i < tapFields.length; i++) {
+            setValue(`${arrayFormName}[${i}].${SELECTED}`, true);
+        }
+    }
+
+    function unselectAllRows() {
+        for (let i = 0; i < tapFields.length; i++) {
+            setValue(`${arrayFormName}[${i}].${SELECTED}`, false);
+        }
+    }
+
+    function moveUpSelectedRows() {
+        const currentTapRows = getValues(arrayFormName);
+
+        if (currentTapRows[0][SELECTED]) {
+            // we can't move up more the rows, so we stop
+            return;
+        }
+
+        for (let i = 1; i < currentTapRows.length; i++) {
+            if (currentTapRows[i][SELECTED]) {
+                swap(i - 1, i);
+            }
+        }
+    }
+
+    function moveDownSelectedRows() {
+        const currentTapRows = getValues(arrayFormName);
+
+        if (currentTapRows[currentTapRows.length - 1][SELECTED]) {
+            // we can't move down more the rows, so we stop
+            return;
+        }
+
+        for (let i = currentTapRows.length - 2; i >= 0; i--) {
+            if (currentTapRows[i][SELECTED]) {
+                swap(i, i + 1);
+            }
+        }
+    }
+
+    function onDragEnd(result) {
+        // dropped outside the list
+        if (!result.destination) {
+            return;
+        }
+
+        move(result.source.index, result.destination.index);
+    }
+
+    //TODO fix alignment of column names when no rows ?
```

Comment:
TODO when ?

Reply:
When it will be done for all the application because it's everywhere like that from what I saw